vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -129,11 +129,11 @@ def parse_pod(pod)
       new_result = parse_base_item(pod)
 
       new_result.merge!(
-        :restart_policy => pod.spec.restartPolicy,
-        :dns_policy     => pod.spec.dnsPolicy,
-        :ipaddress      => pod.status.podIP,
-        :container_node => nil,
-        :containers     => []
+        :restart_policy        => pod.spec.restartPolicy,
+        :dns_policy            => pod.spec.dnsPolicy,
+        :ipaddress             => pod.status.podIP,
+        :container_node        => nil,
+        :container_definitions => []
       )
 
       unless pod.spec.host.nil?
```

Comment:
The first thing I would do here is building an index of containers:

      containers_index = {}

      unless pod.status.nil? || pod.status.containerStatuses.nil?
        pod.status.containerStatuses.each do |cn|
          containers_index[cn.name] = parse_container(cn, pod.metadata.uid)
        end
      end